text/0000-standard-dependency-metadata-format.md

```diff
@@ -0,0 +1,48 @@
+# Standardized Paketo Dependency Metadata Format
+
+## Summary
+
+This proposal suggests that the Paketo project should converge on a single dependency metadata format.
+
+## Motivation
+
+There are several projects, such as the introduction of ARM or the removal of stacks, that are being discussed that would force Paketo to update the metadata for dependencies. It seems prudent that if we are going to have to do a large update to dependency metadata that it might also be a good time for us to converge as a project on the dependency metadata format that we should be using going forward.
+
+This could also be a good first step in converging on shared tooling. By having a shared dependency metadata format we would have a good common convergence point to begin building universal tooling for the Paketo project. 
+
+## Implementation
+
+The following in the proposed metadata format:
+
+```toml
+[[metadata.dependencies]]
+arch = "<dependency compatible architecture>"
+checksum = "<dependency algo:checksum>"
+id = "<dependency ID>"
+os = "<dependency compatible OS>"
+uri = "<dependency URI>"
+version = "dependency version"
```

Comment:
I'm worried that this will get tedious. It is entirely possible to have binaries, like JAR files, that are not architecture/OS/distro dependent. The distro is a list so we could have multiple, it's also option so we can leave it off. Arch and OS are required and individual.

Thus how would I represent something that runs on all arch/os? Or what if it runs on two arch's like ARM64 and AMD64, do I then need to duplicate the metadata and include it twice?

Off the top of my head...

- If we have arch and OS as optional, then we could specify that the lack of them indicates support for all
- If we make them arrays, then we can specify support for multiple items.

I'm open to other ideas. Thoughts?

Reply:
>If we have arch and OS as optional, then we could specify that the lack of them indicates support for all

+1

Reply:
> I think I would ask this question a different way - how are you planning on handling the stack removal (replacing it with os etc above).

Not sure if this was addressed to me or @ForestEckhardt.

I haven't thought about it too much. The upstream is not ready for it yet, and it's missing some functionality we need (there's another RFC proposed to fill this gap, but we'd need this or a similar RFC to remove stacks).

I would say that my goal is to limit repetition and to keep the TOML reasonable to read. That was my question here. The way things are proposed currently, it would require a lot of repeated TOML for various dependencies. We need to find a more efficient way to represent that in the metadata. My $0.02.

Reply:
I think that I am more than happy with making the OS and Arch optional (and will do so in my next commit)